src/main/java/uk/gov/hmcts/reform/pip/data/management/models/publication/HeaderGroup.java

```diff
@@ -83,4 +85,23 @@ public class HeaderGroup {
     @Valid
     private LocalDateTime contentDate;
 
+    /**
+     * Returns a map of the headers used to log within app insights.
+     * @return A map of headers
+     */
+    public Map<String, String> getAppInsightsHeaderMap() {
+        Map<String, String> map = new ConcurrentHashMap<>();
+        map.computeIfAbsent("PROVENANCE", val -> provenance);
+        map.computeIfAbsent("SOURCE_ARTEFACT_ID", val -> sourceArtefactId);
+        map.computeIfAbsent("TYPE", val -> type.toString());
+        map.computeIfAbsent("SENSITIVITY", val -> sensitivity.toString());
+        map.computeIfAbsent("LANGUAGE", val -> language.toString());
+        map.computeIfAbsent("DISPLAY_FROM", val -> displayFrom.toString());
```

Comment:
The toString() method is going to output the string in the format like below. If that is what you want to show in appInsights then it is fine. If you prefer to output it using a specific date format might be better to use displayFrom.format(DateTimeFormatter.ofPattern({XXX}). This is the same for all date fields.

[2022, 12, 14, 17, 30, 00, 000000]

Reply:
hmm from testing it toString() appears to output it in app insights like: 2022-11-22T12:33:55.1070000Z

Reply:
Ohh cool that is fine then. Will close this one :)